src/batch/src/executor/join/lookup_join.rs

```diff
@@ -0,0 +1,576 @@
+                        JoinType::LeftOuter if !chunk_is_none => {
+                            self.do_inner_join(&cur_row, probe_side_chunk)
+                        }
+                        _ => Err(ErrorCode::NotImplemented(
+                            format!(
+                                "Lookup Join does not support join type {:?}",
+                                self.join_type
+                            ),
+                            None.into(),
+                        )
+                        .into()),
+                    }?;
+
+                    // Append chunk from the join result to the chunk builder if it exists
+                    if let Some(return_chunk) = join_result {
+                        if return_chunk.cardinality() > 0 {
+                            // println!{"{:?}", return_chunk}
+                            chunk_added = true;
+
+                            if let Some(inner_chunk) =
+                                self.append_chunk(SlicedDataChunk::new_checked(return_chunk)?)?
+                            {
+                                yield inner_chunk.reorder_columns(&self.output_indices);
+                            }
+                        }
+                    }
+
+                    // Until we don't have any more last_chunks to append to the chunk builder,
+                    // keep appending them to the chunk builder
+                    while self.last_chunk.is_some() {
+                        let mut temp_chunk: Option<SlicedDataChunk> = None;
+                        std::mem::swap(&mut temp_chunk, &mut self.last_chunk);
+                        if let Some(inner_chunk) = self.append_chunk(temp_chunk.unwrap())? {
+                            yield inner_chunk.reorder_columns(&self.output_indices);
+                        }
+                    }
+                }
+            }
+        }
+
+        // Consume and yield all the remaining chunks in the chunk builder
+        if let Some(data_chunk) = self.chunk_builder.consume_all()? {
+            yield data_chunk.reorder_columns(&self.output_indices);
+        }
+    }
+
+    /// Inner Joins the `cur_row` with the `probe_side_chunk`. The non-equi condition is also
+    /// evaluated to hide the rows in the join result that don't match the condition.
+    fn do_inner_join(
+        &self,
+        cur_row: &RowRef,
+        probe_side_chunk: Option<Result<DataChunk>>,
+    ) -> Result<Option<DataChunk>> {
+        if let Some(probe_chunk) = probe_side_chunk {
+            let probe_chunk = probe_chunk?.compact()?;
+
+            let const_row_chunk =
+                convert_row_to_chunk(cur_row, probe_chunk.capacity(), &self.build_side_data_types)?;
+
+            let new_chunk = concatenate(&const_row_chunk, &probe_chunk)?;
+
+            if let Some(cond) = self.condition.as_ref() {
+                let visibility = cond.eval(&new_chunk)?;
```

Comment:
Expression has `eval_row` now, I think we can use that.

Reply:
`new_chunk` can have more than one row in it, so I don't think using `eval_row` is possible.